### PR TITLE
feat(Toggle): adds typescript typings to Toggle

### LINKED ISSUE #12564: Add TypeScript types to Toggle

### PR DESCRIPTION
**Pull request by erifsx**
Closes #12564

Adds typescript typing to the Toggle component

#### Changelog

**New**
- Copies the typings for Tooltip from the definitely typed repo and updates to match current typings.

**Changed**
- the jsDoc for useControllableState had onChange as a required field and the jsDoc now types it as optional

#### Testing / Reviewing

- should build
- storybooks should work
- all existing fields should be accessible from typescript

### CODE REVIEW

**Comment by erifsx** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by erifsx** [comment]
recheck

File: packages/react/src/components/Toggle/Toggle.tsx [hunk 1]
```diff
@@ -1,16 +1,96 @@
...
+  /**
+   * If true, the side labels (props.labelA and props.labelB) will be replaced by
+   * props.labelText (if passed), so that the toggle doesn't render a top label.
+   */
+  hideLabel?: boolean;
+
+  /**
+   * Provide an event listener that is called when the control is toggled
+   */
+  onClick: MouseEventHandler<HTMLDivElement> | undefined;
```
Comment (francinelucca) [inline packages/react/src/components/Toggle/Toggle.tsx]: looks like this function also gets called when the button is activated via keyboard? in that case we need to adjust the type
```suggestion
  onClick: MouseEventHandler<HTMLDivElement> | MouseEventHandler<HTMLButtonElement> | undefined;
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
Thanks for putting this PR together @erifsx ! mostly LGTM, just one minor change request 🙏🏻.
TIA

File: packages/react/src/components/Toggle/Toggle.tsx (hunk 1 above)
Comment (erifsx) [reply packages/react/src/components/Toggle/Toggle.tsx]: Thanks, good catch! 

Added typings for both mouse and keyboard events.

**Review by tay1orjones — APPROVED** [review]

**Review by francinelucca — APPROVED** [review]

**Pull request merged**